server/graphql/general/src/queries/insurances.rs

```diff
@@ -0,0 +1,189 @@
+use async_graphql::{Context, Enum, InputObject, Object, Result, SimpleObject, Union};
+use chrono::NaiveDate;
+use graphql_core::{
+    generic_filters::EqualFilterStringInput,
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use repository::{
+    name_insurance_join_row::{
+        InsurancePolicyType, NameInsuranceJoinFilter, NameInsuranceJoinRow, NameInsuranceJoinSort,
+        NameInsuranceJoinSortField,
+    },
+    EqualFilter,
+};
+use serde::Serialize;
+use service::auth::{Resource, ResourceAccessRequest};
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+#[graphql(rename_items = "camelCase")]
+pub enum InsuranceSortFieldInput {
+    ExpiryDate,
+    IsActive,
+}
+
+#[derive(InputObject)]
+pub struct InsuranceSortInput {
+    /// Sort query result by `key`
+    key: InsuranceSortFieldInput,
+    /// Sort query result is sorted descending or ascending (if not provided the default is
+    /// ascending)
+    desc: Option<bool>,
+}
+
+#[derive(InputObject, Clone)]
+pub struct InsuranceFilterInput {
+    pub insurance_provider_id: Option<EqualFilterStringInput>,
+}
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq, Debug, Serialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum InsurancePolicyNodeType {
+    Personal,
+    Business,
+}
+
+impl InsurancePolicyNodeType {
+    pub fn from_domain(policy_type: &InsurancePolicyType) -> InsurancePolicyNodeType {
+        use InsurancePolicyType::*;
+        match policy_type {
+            Personal => InsurancePolicyNodeType::Personal,
+            Business => InsurancePolicyNodeType::Business,
+        }
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub struct InsuranceNode {
+    insurance: NameInsuranceJoinRow,
+}
+
+#[Object]
+impl InsuranceNode {
```

Comment:
Realised that pattern follows `row()` rather than the node name. Will update after review